### PR TITLE
add onclose metadata to docs

### PR DESCRIPTION
**Pull request by drewbolles**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,9 +57,45 @@ Easily access a user's insurance information using Trellis Connect.
       // Called when the user closes the widget -- either when they have
       // successfully loaded their policies (potentially after an onSuccess() call) or by
       // clicking the "X" button in the top right of the widget.
+      //
+      // Metadata:
+      //   reason: The action the user took to close the widget. One of the following values.
+      //     * Clicked browser back button
```
Comment (zachwills) [inline README.md:63]: @drewbolles can we use an enum for this so there is more confidence for implementers that these values won't change? What do you think?

File: README.md [hunk 2]
```diff
@@ -57,9 +57,45 @@ Easily access a user's insurance information using Trellis Connect.
...
+      //     * Clicked close button
+      //     * Clicked exit button
+      //     * Completed Connect flow
+      //     * Existing customer
+      //
+      //   status: The point at which the user exited the Link flow. One of the following values.
```
Comment (zachwills) [inline README.md:69]: In their current form I think these really should be `phase` or `screen`. We could add a `status` with `enums` that are a bit more abstracted.

File: README.md (hunk 1 above)
Comment (drewbolles) [reply README.md:63]: By the technical definition of enum, this is that. A specific set of enumerated values for reason. I believe you are more referring to the casing be in snake case, correct?

Comment (zachwills) [reply README.md:63]: Yea - them being more obviously set constants vs. strings that might change randomly.

File: README.md (hunk 2 above)
Comment (drewbolles) [reply README.md]: I updated this property to be `view_name` this matches the metadata for the `TRANSITION_VIEW` event that we provide in the `onEvent` callback. This property seems to be the same, and now behaves in the same manner.

File: README.md [hunk 3]
```diff
@@ -57,9 +57,25 @@ Easily access a user's insurance information using Trellis Connect.
       // Called when the user closes the widget -- either when they have
       // successfully loaded their policies (potentially after an onSuccess() call) or by
       // clicking the "X" button in the top right of the widget.
+      //
+      // Metadata:
+      //   status: The point at which the user exited the Link flow. One of the following values.
```
Comment (zachwills) [inline README.md:62]: @drewbolles can we add descriptions explaining when a dev would see each of these?

Comment (zachwills) [inline README.md:62]: Exited the `connect` flow.

File: README.md [hunk 4]
```diff
@@ -57,9 +57,25 @@ Easily access a user's insurance information using Trellis Connect.
...
+      //     * issuer_selection - User prompted to select an issuer
+      //     * issuer_not_found - User was unable to find their issuer
+      //     * unsupported_issuer - User selected an unsupported issuer
+      //     * trellis_consent - User prompted to accept Trellis' consent
+      //     * has_online_login - User prompted to answer if they have an online login for their issuer
+      //     * create_account - User prompted to create a login if they responded "no" to having an online login
+      //     * requires_credentials - User prompted to provide credentials for the selected issuer
+      //     * requires_questions - User prompted to answer security questions
```
Comment (zachwills) [inline README.md:70]: `security questions to login`

File: README.md [hunk 5]
```diff
@@ -57,9 +57,25 @@ Easily access a user's insurance information using Trellis Connect.
...
+      //     * requires_selections - User prompted to answer multiple choice question(s)
```
Comment (zachwills) [inline README.md:71]: `...to login`

File: README.md [hunk 6]
```diff
@@ -57,9 +57,25 @@ Easily access a user's insurance information using Trellis Connect.
...
+      //     * requires_code - User prompted to provide a one-time passcode
```
Comment (zachwills) [inline README.md:72]: `...to login`

**Pull request merged**